src/RequestHeaderParser.php

```diff
@@ -15,15 +15,16 @@
 class RequestHeaderParser extends EventEmitter
 {
     private $buffer = '';
-    private $maxSize = 4096;
+    private $bufferMaxSize;
 
     private $localSocketUri;
     private $remoteSocketUri;
 
-    public function __construct($localSocketUri = null, $remoteSocketUri = null)
+    public function __construct($localSocketUri = null, $remoteSocketUri = null, $bufferMaxSize = null)
```

Comment:
IMO it should be called `$maxHeaderSize` or `$maxSize`, the buffer might be larger than the headers are if there's a lot of data incoming at the same time.